### PR TITLE
Feature: Phalanx Check in Galaxyview & Teamview Status Page

### PR DESCRIPTION
**Pull request by kejith**
With this PR, a feature is added that allows the galaxy view and status page of Teamview to display the ranges of certain phalanxes.

Galaxy View
image

Teamview Status Page
image

Future Implementations planned
- Tooltip for Teamview Status Page to see which Phalanxes have the System in Range
- Add a List of Phalanxes beneath the Status Page (like the List in Planets/Reports)

Because I was a little bit careless and worked on the same branch as the PR i  accidently added the DB migration functionality and the setup for Typescript to this PR. So, yeah. YAY

`npx knex migrate:latest` should be run to apply DB changes. Maybe someone can come up with a Solution to run this Command after a Devcontainer build/Docker Compose Build?

### CODE REVIEW

**Comment by anoff** [comment]
> npx knew migrate:latest should be run to apply DB changes. Maybe someone can up with a Solution to run this Command after a Devcontainer build/Docker Compose Build?

Maybe `postStartCommand`, see https://containers.dev/implementors/json_reference/

**Comment by kejith** [comment]
> > npx knew migrate:latest should be run to apply DB changes. Maybe someone can up with a Solution to run this Command after a Devcontainer build/Docker Compose Build?
> 
> Maybe `postStartCommand`, see https://containers.dev/implementors/json_reference/

Added the DB-Migration Process to docker build and run process. I also added an SQL-Query that will be run over the existing reports table to pre populate the phalanxes table.

So now this patch should be working by just pulling and opening the devcontainer or running docker compose.

File: userscript/features/galaxyview.js [hunk 1]
```diff
@@ -355,14 +355,48 @@ function modifyTable (data, modfiyFn) {
...
+    })
+}
+
 function init () {
   if (!(window.location.search.includes('page=galaxy') && !window.location.search.includes('tv=station'))) return
   addColumn(2, ['Player Stats', 'Spio Info'])
   addUploadSection()
   modifyTable({}, modifyAddRankFromPopup)
   const data = getVisibleSystem()
+  console.log({ data })
```
Comment (anoff) [inline userscript/features/galaxyview.js:391]: can we get rid of this or wrap it in a `if (teamviewDebug)`

**Review by anoff — CHANGES_REQUESTED** [review]
what about dat log eh?

File: userscript/features/galaxyview.js (hunk 1 above)
Comment (kejith) [reply userscript/features/galaxyview.js:391]: OH NO, shame on me. PR incoming.

**Review by anoff — APPROVED** [review]

**Comment by anoff** [comment]
THANKS FOR YOUR CONTRIBUTION THIS LOOKS NICE (NOW THAT YOU REMOVED THE LOG)

**Pull request merged**